internal/k8s/controller.go

```diff
@@ -2372,14 +2372,24 @@ func (lbc *LoadBalancerController) getServiceForIngressBackend(backend *extensio
 
 // IsNginxIngress checks if resource ingress class annotation (if exists) is matching with ingress controller class
 // If annotation is absent and use-ingress-class-only enabled - ingress resource would ignore
-func (lbc *LoadBalancerController) IsNginxIngress(ing *extensions.Ingress) bool {
-	if class, exists := ing.Annotations[ingressClassKey]; exists {
-		if lbc.useIngressClassOnly {
-			return class == lbc.ingressClass
+func (lbc *LoadBalancerController) IsNginxIngress(obj interface{}) bool {
```

Comment:
this slightly doesn't make sense to me why we take either string or `*extensions.Ingress` as argument, because they are completely different things.

 Would it make it simpler if we take  Ingress,  VirtualServer and VirtualServerRoute as arguments?

Reply:
Not sure it would make it simpler to pass three arguments instead of one. In the end, we're just checking the string even if it's in an annotation, right?

Reply:
I made a mistake: used `and` instead of `or` :( so we still take the obj as an argument, but that obj can be Ingress, VS or VSR.

 What I meant is something like that:
```
	var class string
	switch obj.(type) {
	case *conf_v1.VirtualServerRoute:
		. . .
        case  *conf_v1.VirtualServer:
                . . .
	case *extensions.Ingress:
		ing := obj.(*extensions.Ingress)
		if ingClass, exists := ing.Annotations[ingressClassKey]; exists {
			class = ingClass
		}
```

Or alternatively, when you call this function, you call for VS and VRS like this: 
```
lbc.HasCorrectIngressClass(vs.Spec.IngressClass)
```
So why not do the same for Ingress resource
```
lbc.HasCorrectIngressClass(ing.Annotations[ingressClassKey])
```